network/nodes.py

```diff
@@ -152,6 +152,9 @@ def _add_supplemental_account_information(self, network, main_public_key, json_n
 
 		if not self.is_nem:
 			json_node['extraData']['finalizedHeight'] = api_client.get_finalization_info().height
+			json_node['extraData']['finalizedEpoch'] = api_client.get_finalization_info().epoch
```

Comment:
also here, is there a reason to query the node multiple times instead of just caching the finalization info?

Reply:
You are right, I already cached the finalization info.